### ISSUE TITLE
BUG: np.percentile(..., interpolation='midpoint') seems to have just broken

### ISSUE THREAD

**Issue by jreback** [numpy/numpy#7163]
These matched here: ``1.12.0.dev0+0bba665``

now
```
In [22]: np.__version__
Out[22]: '1.12.0.dev0+d1dada1'

In [19]: arr = np.array([0,2,4])

In [20]: np.percentile(arr,[25,50],interpolation='linear')
Out[20]: array([ 1.,  2.])

In [21]: np.percentile(arr,[25,50],interpolation='midpoint')
Out[21]: array([ 1.,  2.])
```

for comparison to 10.2
```
In [5]: np.__version__
Out[5]: '1.10.2'

In [1]: In [19]: arr = np.array([0,2,4])

In [2]: In [20]: np.percentile(arr,[25,50],interpolation='linear')
Out[2]: array([ 1.,  2.])

In [4]: In [21]: np.percentile(arr,[25,50],interpolation='midpoint')
Out[4]: array([ 1.,  3.])
```

**Comment by charris** [comment]
The development version is a bug fix, see #7129. It hasn't been backported to 1.11, but could be it that seems desirable.

**Comment by jreback** [comment]
thanks @charris I worked around
